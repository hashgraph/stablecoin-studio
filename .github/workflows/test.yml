name: test
on:
  pull_request:
    branches: [ main, release/**, feature/** ]
  push:
    branches: [ main, release/** ]
    tags: [ v* ]
jobs:
  test-node:
    name: Unit Tests
    runs-on: ubuntu-latest
    steps:
      - name: Use Node.js TLS 16.x
        uses: actions/setup-node@v1
        with:
          node-version: 16.x
      - name: Checkout repository
        uses: actions/checkout@v2
      - name: Install dependencies
        run: npm i
      - name: Build
        run: npm run build
      - name: Test
        run: npm run test