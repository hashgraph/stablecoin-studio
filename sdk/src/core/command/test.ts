/* eslint-disable @typescript-eslint/no-empty-interface */

interface CommandResponse {}
interface ICommand {}

export class Command<T> implements ICommand {
	private $resultType!: T;
}

export interface ICommandBus {
	execute<X>(command: Command<X>): Promise<X>;
}

export type IInferringCommandHandler<CommandType extends Command<unknown>> =
	CommandType extends Command<infer ResultType>
		? {
				execute(command: CommandType): Promise<ResultType>;
		  }
		: never;

class ConcreteCommandResponse implements CommandResponse {
	constructor(public readonly status: string) {}
}

class ConcreteCommand extends Command<ConcreteCommandResponse> {
	constructor(public readonly value: string) {
		super();
	}
}

class CommandHandler<CommandType extends Command<unknown>>
	implements IInferringCommandHandler<ConcreteCommand>
{
	execute(command: ConcreteCommand): Promise<ConcreteCommandResponse> {
		throw new Error('Method not implemented.');
	}
}

class CommandBus implements ICommandBus {
	execute<X>(command: Command<X>): Promise<X> {
		throw new Error('Method not implemented.');
	}
}

// Test

const bus = new CommandBus();
const res = bus.execute(new ConcreteCommand('1'));
