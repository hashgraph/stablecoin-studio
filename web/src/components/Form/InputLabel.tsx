import type { FormLabelProps as ChakraInputProps } from '@chakra-ui/react';
import { FormLabel, HStack, Text } from '@chakra-ui/react';
import type { ReactNode } from 'react';

export interface InputLabelProps {
	isRequired: boolean;
	style?: ChakraInputProps;
	children: ReactNode;
}

const InputLabel = ({ isRequired, style, children }: InputLabelProps) => {
	return (
		<FormLabel {...style}>
			<HStack>
				<Text>{children}</Text>
				{isRequired && (
					<Text color='brand.red' data-testid='required'>
						*
					</Text>
				)}
			</HStack>
		</FormLabel>
	);
};

export default InputLabel;
