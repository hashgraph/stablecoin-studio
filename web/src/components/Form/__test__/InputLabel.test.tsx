import { render } from '../../../test/index';
import type { InputLabelProps } from '../InputLabel';
import InputLabel from '../InputLabel';

const inputLabelProps: InputLabelProps = {
	children: 'Label',
	isRequired: false,
};

describe(`<${InputLabel.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<InputLabel {...inputLabelProps} />);

		expect(component.asFragment()).toMatchSnapshot();
	});
	test('should add * if it is required', () => {
		const component = render(<InputLabel {...inputLabelProps} isRequired={true} />);

		expect(component.getByTestId('required')).toBeInTheDocument();
	});

	test('should not show * if it is not required', () => {
		const component = render(<InputLabel {...inputLabelProps} />);

		expect(component.queryByTestId('required')).toBeNull();
	});
});
