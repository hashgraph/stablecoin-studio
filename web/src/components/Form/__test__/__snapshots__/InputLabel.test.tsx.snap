// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputLabel /> should render correctly 1`] = `
<DocumentFragment>
  <label
    class="chakra-form__label css-1hfoxt1"
  >
    <div
      class="chakra-stack css-1igwmid"
    >
      <p
        class="chakra-text css-mgupmj"
      >
        Label
      </p>
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
